packages/cozy-mespapiers-lib/src/components/Search/search.js

```diff
@@ -78,43 +72,47 @@ export const makeContactTags = contact => {
   return contactTags
 }
 
-export const addFileDoc = (index, doc, scannerT) => {
-  return index.add({
-    ...doc,
-    flexsearchProps: {
-      tag: makeFileTags(doc),
-      name: doc.name,
-      translatedQualificationLabel: scannerT(
-        `items.${doc.metadata.qualification.label}`
-      )
-    }
-  })
+export const addFileDoc = (index, doc, t) => {
+  if (hasQualifications(doc)) {
+    return index.add({
+      ...doc,
+      _type: doc._type || doc.type || FILES_DOCTYPE, // should be useless after solving https://github.com/cozy/cozy-libs/issues/2023
+      flexsearchProps: {
+        tag: makeFileTags(doc),
+        translatedQualificationLabel: t(
+          `items.${doc.metadata.qualification.label}`
+        )
+      }
+    })
+  }
 }
 
 export const addContactDoc = (index, doc) => {
   return index.add({
     ...doc,
+    _type: doc._type || doc.type || CONTACTS_DOCTYPE, // should be useless after solving https://github.com/cozy/cozy-libs/issues/2023
     flexsearchProps: {
       tag: makeContactTags(doc),
-      fullname: doc.fullname,
-      birthday: doc.birthday,
-      birthcity: doc.birthcity,
-      email: doc.email?.[0]?.address,
-      civility: doc.civility,
-      company: doc.company,
-      jobTitle: doc.jobTitle
+      email: doc.email?.[0]?.address
     }
   })
 }
 
-export const addDocs = ({ index, docs, scannerT }) => {
-  for (const doc of docs) {
-    if (isFile(doc)) {
-      addFileDoc(index, doc, scannerT)
-    }
+export const addDoc = ({ index, doc, t }) => {
+  if (isFile(doc)) {
+    addFileDoc(index, doc, t)
+  } else {
+    // ⚠️ io.cozy.contacts is implicit here, we should solve https://github.com/cozy/cozy-libs/issues/2023
+    addContactDoc(index, doc)
+  }
+}
 
-    if (doc._type === CONTACTS_DOCTYPE) {
-      addContactDoc(index, doc)
-    }
+// We do not store the documents in the flexsearch store, only the _id and _type.
+// To retrieve the documents, we make a link with the _id.
+// So there is no update of the index to do, only additions and deletions.
```

Comment:
Actually, I realize it is necessary to update the index for an update: even if the stored content is simply the (_id, _type), the searched terms might have been changed. 
Imagine a file rename from "foo" to "bar". The associated id didn't change, but we have to update the index to be able to search on "bar"